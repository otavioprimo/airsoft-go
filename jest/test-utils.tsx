import React from 'react';
import {render} from '@testing-library/react-native';
import {ThemeProvider} from 'styled-components/native';
import {createStore, Reducer} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import * as Theme from '../src/theme';

interface ReduxProviders {
  state: any;
  reducer: Reducer<any, any>;
}

interface RenderProviders {
  children: React.ReactElement;
  options?: ReduxProviders;
}

const ReduxProvider = ({children, options}: any) => {
  if (!options) {
    return <>{children}</>;
  }

  const store = createStore(options.reducer, options.state);
  return <Provider store={store}>{children}</Provider>;
};

const AllTheProviders: React.FC<RenderProviders> = ({children, options}) => {
  return (
    <ThemeProvider theme={Theme.DarkTheme}>
      <ReduxProvider options={options}>{children}</ReduxProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: ReduxProviders) => {
  return render(<AllTheProviders options={options}>{ui}</AllTheProviders>);
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
