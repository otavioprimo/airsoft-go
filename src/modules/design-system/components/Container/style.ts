import styled from 'styled-components/native';

export const SafeArea = styled.SafeAreaView`
  background-color: ${({theme}) => theme.primary};
`;

export const ScrollView = styled.ScrollView``;
