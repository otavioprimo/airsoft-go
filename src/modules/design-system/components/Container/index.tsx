import React from 'react';

import {ScrollView, SafeArea} from './style';

const Container: React.FC = ({children}) => {
  return (
    <SafeArea>
      <ScrollView>{children}</ScrollView>
    </SafeArea>
  );
};

export default Container;
