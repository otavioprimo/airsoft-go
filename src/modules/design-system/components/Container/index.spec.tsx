import React from 'react';
import {act} from 'react-test-renderer';
import {render} from 'test/test-utils';
import {Text, View} from 'react-native';

import Container from './index';

afterEach(async () => {
  await act(async () => {});
});

describe('Design System - Container', () => {
  it('Should match snapshot', () => {
    const {toJSON} = render(<Container />);
    expect(toJSON()).toMatchSnapshot('DS_Container');
  });

  it('Should render children', () => {
    const childrenToRender = <Text>My Children Component</Text>;
    const containerWithChildren = <Container>{childrenToRender}</Container>;
    const {getByText} = render(containerWithChildren);
    const textComponent = getByText('My Children Component');
    expect(textComponent).toBeDefined();
  });

  it('Should render children with children', () => {
    const childrenToRender = <Text>My Children Component</Text>;
    const containerWithChildren = (
      <Container>
        <View>{childrenToRender}</View>
      </Container>
    );
    const {getByText} = render(containerWithChildren);
    const textComponent = getByText('My Children Component');
    expect(textComponent).toBeDefined();
  });
});
