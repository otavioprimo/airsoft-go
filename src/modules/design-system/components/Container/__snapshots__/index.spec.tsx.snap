// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System - Container Should match snapshot: DS_Container 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#222",
      },
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {},
      ]
    }
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
